system/security/cryptohelper/pke.cpp

```diff
@@ -0,0 +1,225 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+/*
+ * Module dealing with openssl asymmetric key encryption/decription.
+ * For now just RSA.
+ */
+
+#if defined(_USE_OPENSSL) && !defined(_WIN32)
+
+#include "jliball.hpp"
+#include <openssl/pem.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+
+#include "cryptocommon.hpp"
+#include "pke.hpp"
+
+namespace cryptohelper
+{
+
+void CLoadedKey::loadKeyBio(const char *keyMem)
+{
+    keyBio.setown(BIO_new_mem_buf(keyMem, -1));
+    if (!keyBio)
+        throwEVPException(0, "loadKeyBio: Failed to create bio for key");
+}
+
+bool CLoadedKey::loadKeyFileToMem(MemoryBuffer &keyMb, const char *keyFile)
+{
+    OwnedIFile iFile = createIFile(keyFile);
+    OwnedIFileIO iFileIO = iFile->open(IFOread);
+    if (!iFileIO)
+        return false;
+    size32_t sz = iFile->size();
+    verifyex(read(iFileIO, 0, iFile->size(), keyMb) == sz);
```

Comment:
minor: should use sz instead of iFile->size()

Reply:
agree, will change.